src/main/java/org/example/Main.java

```diff
@@ -0,0 +1,97 @@
+package org.example;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+import java.util.Objects;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.Map;
+import java.time.LocalDate;
+
+public class Main {
+    public static void main(String[] args) {
+        List<Package> packages = new ArrayList<>();
+
+        try {
+            Scanner scanner = new Scanner(new File("test-data.txt"));
+
+            while (scanner.hasNextLine()) {
+                String line = scanner.nextLine();
+                String[] fields = line.split(",");
+
+                String targetLocation = fields[0].trim();
+                int targetDistance = Integer.parseInt(fields[1].trim());
+                double packageValue = Double.parseDouble(fields[2].trim());
+                LocalDate deliveryDate = LocalDate.parse(fields[3].trim());
+
```

Comment:
make all the file reading in a separate method, same for mapping to java objects